src/mol-plugin/state.ts

```diff
@@ -89,6 +92,9 @@ class PluginState extends PluginComponent {
         if (snapshot.structureFocus) {
             this.plugin.managers.structure.focus.setSnapshot(snapshot.structureFocus);
         }
+        if (snapshot.structureSelection) {
+            this.plugin.managers.structure.selection.setSnapshot(snapshot.structureSelection);
```

Comment:
Will this work on fresh snapshots where the visual isn't created yet?

Reply:
It needs the structures the bundles are refering to in the state tree which is the case here with the data snapshot applied before. Visuals can be created later and get the selection mark applied like normally. Also it worked for me using it :)